scan/lib/scan/runner.rb

```diff
@@ -126,7 +126,7 @@ def retryable_tests(input)
 
         test_cases = suite.split(":\n").fetch(1, []).split("\n").each
                           .select { |line| line.match?(/^\s+/) }
-                          .map { |line| line.strip.gsub(".", "/").gsub("()", "") }
+                          .map { |line| line.strip.gsub(/[\s\.]/, "/").gsub(/[\-\[\]\(\)]/, "") }
```

Comment:
The logic of the first regexp: if there exists `.` or white spaces, then replace them with `/`.
The logic of the second regexp: if there exists `-[]()`, then erase them. It is impossible for a single legal `-`, `[`, `(` or `)` to appear in the test name so we simply use a pattern that matches every single occurrence of them, rather than matching the sequence of `-[` and `()`.